src/clr-angular/data/tree-view/recursive-for-of.ts

```diff
@@ -28,17 +30,30 @@ export class ClrRecursiveForOf<T> implements OnChanges {
   // TODO: accept NgIterable<T> return type
   @Input('clrRecursiveForGetChildren') getChildren: (node: T) => AsyncArray<T>;
 
+  private subscriptions: Subscription[] = [];
+
   // I'm using OnChanges instead of OnInit to easily keep up to date with dynamic trees. Maybe optimizable later.
   ngOnChanges() {
     let wrapped: RecursiveTreeNodeModel<T>[];
     if (Array.isArray(this.nodes)) {
-      wrapped = this.nodes.map(node => new RecursiveTreeNodeModel(node, null, this.getChildren));
+      wrapped = this.nodes.map(node => new RecursiveTreeNodeModel(node, null, this.getChildren, this.featuresService));
     } else {
-      wrapped = [new RecursiveTreeNodeModel(this.nodes, null, this.getChildren)];
+      wrapped = [new RecursiveTreeNodeModel(this.nodes, null, this.getChildren, this.featuresService)];
     }
+    this.subscriptions.push(
```

Comment:
Can we leave a note here that this may be a performance issue if the children inputs change a lot, or add a guard here?

Reply:
Guarded!